src/group_theory/perm/cycles.lean

```diff
@@ -166,6 +166,18 @@ calc sign f = sign (swap x (f x) * (swap x (f x) * f)) :
         pow_one, units.neg_mul_neg] }
 using_well_founded {rel_tac := λ _ _, `[exact ⟨_, measure_wf (λ f, f.support.card)⟩]}
 
+lemma is_cycle_of_is_cycle_pow {σ : perm α} {n : ℤ}
+  (h1 : is_cycle (σ ^ n)) (h2 : σ.support.card ≤ (σ ^ n).support.card) : is_cycle σ :=
```

Comment:
I'm not so convinced by the phrasing of `h2`: why not just `σ.support = (σ ^ n).support`? Could you give an example where `card ≤ card` is nicer to work with?

Reply:
Here's an example of `σ.support ≤ (σ ^ n).support` being slightly useful (although without the `.card:)` Let `g` be a cycle, and let `n` be coprime to the order of `g`. By the results of #6875, we have `(g ^ n) ^ m = g` for some `m`. By the results of #6872, we have `g.support = ((g ^ n) ^ m).support ≤ (g ^ n).support`. Then by this PR, `g^n` is also a cycle.

It is true that once you know `σ.support.card ≤ (σ ^ n).support.card` or `σ.support ≤ (σ ^ n).support`, it's not too much work to use #6872 to get `σ.support = (σ ^ n).support`. So I would be fine changing to equality (maybe with a comment mentioning that `support_pow_le` is useful).

Reply:
I'm agnostic about the `le` vs `eq` question, but let's drop the `card`.